### DISCUSSION TITLE
Qual o problema da simplicidade no JavaScript?

### DISCUSSION THREAD

**Discussion by RatoX** [frontendbr/forum#317] [Discussão]
Véi, na boa, o que está acontecendo com o JS?

Faz meio que um tempo que estou parado no Javascript, tipo 9 meses, voltei esses dias pensando em modificar por completo o [Marighella](https://github.com/marighella). Comecei pelo backend, servidores e algumas novas estratégias novas que ainda não estão publicas, quando cheguei no frontend tomei o maior susto, temos todo um gigantesco mar de bibliotecas porém parece que tudo tem que resumir entre React ou Angular2 e estou não estou conseguindo levar as coisas por modulos, por exemplo:

1 - Preciso de obter dados por REST; (beleza boto o fetch pra rodar)
2 - Preciso renderizar esses com um sistema de template simples; (sem nenhuma ideia aqui)
3 - Um sistema de autenticação; (OAuth.io)
4 - Um WYSIWYG maneiro; (sem ideias)

Rola umas dicas aqui galera?

**Comment by felipefialho** [comment]
@RatoX Você leu o artigo [How it feels to learn JavaScript in 2016](https://hackernoon.com/how-it-feels-to-learn-javascript-in-2016-d3a717dd577f#.x0t5gkaf2)?

Vai totalmente de acordo ao seu problema 😄 

Realmente, estamos tendo uma overdose de bibliotecas para resolver problemas que nós nem sabíamos que tínhamos, mas não vejo isso como um problema exatamente.

Temos milhares de escolhas antes de iniciar o projeto, mas tudo pode ser resolvido com um pouco de planejamento. E algumas perguntas. 
- Qual o tamanho do projeto?
- Precisa de escalabilidade intensa?
- Qual vai ser o volume de acessos?
- Performance é extremamente importante?
- Qual o tempo de desenvolvimento?

Realmente não acho que é necessário uma metodologia e desenvolvimento em alto nível para por exemplo, um hotsite que vai ficar 6 meses no ar ou um projeto de nicho que vai ter uma quantidade minima de acesso, nesses casos, até o jQuery serve (podem me jogar pedras 😈 ).

Temos que ter em mente sempre, que nosso trabalho é gerar valor e retornar lucro para o produto. Simples assim, as ferramentas são só meios de atingir esses objetivos, e não o objetivo final.

---

Sobre sua dúvida, ajuda nóis aqui @frontendbr/admins

**Comment by lagden** [comment]
eu li esse artigo hoje cedo...
e adoro essa guerra!!!
adoro ver programador backend blowing your fucking mind tentando entender como manipular algo tão livre, não tipado e simples...
adoro ver as grandes empresas criando soluções novas que não funcionam no IE...
adoro coffeescript...
TS é pra backend programar JS...
Babel é a nossa máquina do tempo...

I 💟 JS

**Comment by lagden** [comment]
by the way... não tem nenhum problema... apenas muitas soluções!!!

**Comment by wmartins** [comment]
@LFeh, que baita artigo!

@RatoX, só complementando a resposta do @LFeh, acho que a comunidade começou a pensar nos pacotes de uma forma bem mais "modular", e não só para pacotes, mas para todos os componentes da arquitetura de uma aplicação. Por exemplo, existe um toolkit de CSS - o [Tachyons](http://tachyons.io/docs/) - que te permite instalar cada pedacinho do toolkit como um módulo separado. Por um lado isso é bom, visto que te permite já escolher somente as partes necessárias para construír a tua aplicação. Por outro, isso te leva a ter que ter um tempo maior para analisar cada um desses pedaços e tomar a decisão de usá-lo ou não.

> "parece que tudo tem que resumir entre React ou Angular2"

Entendo o que tu quer dizer e concordo. Acho que o que acontece é que em muitas discussões, "pro caso geral", essas libs/frameworks vão resolver o problema. Porém, acho que o que vale é ler/experimentar cada uma dessas _buzzwords_ (React, Angular, Vue, Elm, Immutable, etc.) e entender qual dessas é a que pode te apresentar melhores resultados.

Se eu posso te dar uma dica, tenta ver isso como um ponto positivo. É um mundo "novo" onde tu não precisa te prender à tecnologia X ou Y para desenvolver algo legal. Tenta analisar quais são as coisas mais "famosas" em uso e experimenta. Daqui a pouco tem alguma que tem uma curva de aprendizado muito acentuada mas te traz um grande ganho, a outra é muito rápida de aprender, mas faz com que muito código tenha que ser produzido.
